quickstarts/microsoft.digitaltwins/digitaltwins-with-function-time-series-database-connection/modules/roleassignment.bicep

```diff
@@ -0,0 +1,67 @@
+@description('Existing Digital Twin resource name')
+param digitalTwinsName string
```

Comment:
change this so that you pass in the principalId of the MSI on the digitalTwins resource - instead of the resourceId.  This will fix idempotency issues with the roleAssignment in the dev/test loop.  Then use it in the roleAssignmentNames below (as appropriate).

Reply:
I'm actually confused by this statement. Can you provide some additional clarity on how to accomplish this?

The digitalTwins resource itself needs certain role assignments to complete the setup; I am not sure I understand why it needs to be done on the MSI.

Reply:
You're in essence relying on the digitaltwins resource having MSI set up properly already, so I think the suggestion is to have an additional parameter for digitalTwins.identity.principalId. However, since we only allow system assigned identities currently, I'm not sure how we would do this.

Reply:
I don't see the reason that would make a difference, if we can just pass the name to get the digitaltwins resource, which will have the principalId anyways

Reply:
As written, the digitalTwins resourceId can be reused with a different principal which would create a conflict on the roleAssigment.  Using the principalId ensures the name will be idempotent and unique

Reply:
Modified the bicep file to accept the principalID of the identity of the digital Twin